cmd/listen-notification-handlers.go

```diff
@@ -116,11 +118,31 @@ func (api objectAPIHandlers) ListenNotificationHandler(w http.ResponseWriter, r
 
 	// Listen Publisher and peer-listen-client uses nonblocking send and hence does not wait for slow receivers.
 	// Use buffered channel to take care of burst sends or slow w.Write()
-	listenCh := make(chan event.Event, globalAPIConfig.getRequestsPoolCapacity()*len(globalEndpoints.Hostnames()))
-
+	mergeCh := make(chan []byte, globalAPIConfig.getRequestsPoolCapacity()*len(globalEndpoints.Hostnames()))
+	localCh := make(chan event.Event, globalAPIConfig.getRequestsPoolCapacity())
+
+	// Convert local messages to JSON and send to mergeCh
+	go func() {
+		var buf bytes.Buffer
+		enc := json.NewEncoder(&buf)
+		tmpEvt := struct{ Records []event.Event }{[]event.Event{{}}}
```

Comment:
```suggestion
		var buf bytes.Buffer
		enc := json.NewEncoder(&buf)
		tmpEvt := struct{ Records []event.Event }{[]event.Event{{}}}
```

you can use `bytebufferpool` here? @klauspost

Reply:
@harshavardhana The `buf bytes.Buffer` is re-used for all writes (this is outside the for loop). But sure, I can init it with a buffer - but it will only save on per RPC call.

Each event uses a pooled buffer. We can avoid the copy, but that would alloc a `bytes.Buffer` and `json.Encoder` for every event.